test-suite/js/src/test/require-2.12/org/scalajs/testsuite/jsinterop/JSOptionalTest212.scala

```diff
@@ -199,14 +199,10 @@ class JSOptionalTest212 {
     assertEquals(Some(3), obj.z)
   }
 
-  @Test def traitWithOptionalFunction(): Unit = {
-    val obj = new TraitWithOptionalFunction {
-      override val f = js.defined(x => x + 1)
-    }
-
-    assertEquals("function", js.typeOf(obj.f))
-    assertEquals(6, obj.f.get(5))
-  }
+  /* @Test def traitWithOptionalFunction(): Unit
+   * Moved to JSOptionalTest212FunParamInference.scala because Scala 3 cannot
+   * infer the parameter type yet when using true union types.
```

Comment:
Reference the relevant dotty issue so whether this is still needed can be easily checked?

Reply:
Good point. Done.